src/main/java/org/opensearch/ad/transport/EntityResultTransportAction.java

```diff
@@ -196,35 +200,51 @@ private ActionListener<Optional<AnomalyDetector>> onGetDetector(
-                if (result.getRcfScore() > 0) {
-                    AnomalyResult resultToSave = result
-                        .toAnomalyResult(
-                            detector,
-                            Instant.ofEpochMilli(request.getStart()),
-                            Instant.ofEpochMilli(request.getEnd()),
-                            executionStartTime,
-                            Instant.now(),
-                            ParseUtils.getFeatureData(datapoint, detector),
-                            categoricalValues,
-                            indexUtil.getSchemaVersion(ADIndex.RESULT),
-                            modelId,
-                            null,
-                            null
-                        );
+                try {
+                    ThresholdingResult result = modelManager
+                        .getAnomalyResultForEntity(datapoint, entityModel, modelId, categoricalValues, detector.getShingleSize());
+                    // result.getRcfScore() = 0 means the model is not initialized
+                    // result.getGrade() = 0 means it is not an anomaly
+                    // So many OpenSearchRejectedExecutionException if we write no matter what
+                    if (result.getRcfScore() > 0) {
+                        AnomalyResult resultToSave = result
+                            .toAnomalyResult(
+                                detector,
+                                Instant.ofEpochMilli(request.getStart()),
+                                Instant.ofEpochMilli(request.getEnd()),
+                                executionStartTime,
+                                Instant.now(),
+                                ParseUtils.getFeatureData(datapoint, detector),
+                                categoricalValues,
+                                indexUtil.getSchemaVersion(ADIndex.RESULT),
+                                modelId,
+                                null,
+                                null
+                            );
 
-                    resultWriteQueue
+                        resultWriteQueue
+                            .put(
+                                new ResultWriteRequest(
+                                    System.currentTimeMillis() + detector.getDetectorIntervalInMilliseconds(),
+                                    detectorId,
+                                    result.getGrade() > 0 ? RequestPriority.HIGH : RequestPriority.MEDIUM,
+                                    resultToSave,
+                                    detector.getResultIndex()
+                                )
+                            );
+                    }
+                } catch (IllegalArgumentException e) {
+                    // fail to score likely due to model corruption. Re-cold start to recover.
+                    cache.get().removeEntityModel(detectorId, modelId);
```

Comment:
Add some todo to track model corruption times?

Reply:
I added the tracking code. Will send the new commit.